### ISSUE TITLE
GitHub enterprise support?

### ISSUE THREAD

**Issue by jasonkarns** [chancancode/blame_parent#3]

**Comment by chancancode** [comment]
I don't have access to the enterprise UI/markup. Is it pretty close to the public one? Guthub "recently" changed the markup and broke the extension. I'll fix it over the weekend, perhaps that would fix this also

**Comment by jasonkarns** [comment]
That's not the problem (I would only expect this to work with recent-ish versions of github enterprise that mirrored similar markup).

The issue is that the manifest restricts the extension to only running on github.com domains. What needs to happen is to add an options page that lets user's add in their own list of urls that the extension can be active for.

See github-notifier for an example: https://github.com/sindresorhus/github-notifier-chrome

(the options page has a url input, and options.js then requests permissions to run on that URL)

**Comment by chancancode** [comment]
Ah. I see! I wouldn't mind supporting that if someone can send a pull request ;)



—

**Comment by jasonkarns** [comment]
Yeah, it would be an awesome feature but isn't a trivial change :)

**Comment by chancancode** [comment]
I fixed the extension! So we have one less blocker for this. It's tricky though, because we are only injecting the script based on the URLs [hardcoded in manifest.json](https://github.com/chancancode/blame_parent/blob/master/manifest.json#L7).

I don't think you can have dynamic values there, so we will have to switch to injecting on all pages by default, but only run the function if it matches a check inside the JavaScript file (which would allow us to read the settings, etc).

**Comment by jasonkarns** [comment]
can't have dynamic values in the content_scripts: matches value, but you don't have to use the content_scripts in the manifest. Content scripts can be injected programatically. The extension would use the permissions API in the manifest, and the valid URLs still stored via options. Then the content scripts are injected programatically: https://developer.chrome.com/extensions/content_scripts#pi

**Comment by jasonkarns** [comment]
I keep intending to spend some time to spike this out. (I maintain a couple of github-related chrome extensions of my own that don't yet support Enterprise) But I haven't done it yet. I'll certainly use sindre's extension as a blueprint.
